core/src/main/java/org/infinispan/persistence/sifs/Compactor.java

```diff
@@ -581,7 +581,8 @@ private void compactSingleFile(int scheduledFile, boolean isLogFile, CompactionE
                   log.tracef("Drop index for key %s, file %d:%d (%s)", key, scheduledFile, scheduledOffset,
                         header.valueLength() > 0 ? "record" : "tombstone");
                }
-               index.handleRequest(IndexRequest.dropped(segment, key, ByteBufferImpl.create(serializedKey), prevFile, prevOffset, scheduledFile, scheduledOffset));
+               aggregateCompletionStage.dependsOn(
```

Comment:
Why is this is required? The indexer is already guaranteeing ordering of submitted tasks. Or is this preventing something else?